### PR TITLE
Question answer

### PR DESCRIPTION
**Pull request by tpminhtam**
- revised the sorting of questions and answers within questions
- added modal to input new questions
- updated server to response to POST request

### CODE REVIEW

File: client/src/components/QuestionsAndAnswers/Answers.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            expanded: !this.state.expanded
+        })
+    }
+
+    render() {
+        const { body, date, helpfulness, photos } = this.props;
+
+        return (
+            <>
+                <div key={body} style={{ display: i < 2 || this.expanded ? 'block' : 'none' }}>
```
Comment (slava000) [inline client/src/components/QuestionsAndAnswers/Answers.jsx]: @tpminhtam what is "i"? where is "i" defined?

File: client/src/components/QuestionsAndAnswers/Answers.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                    <span style={{ fontWeight: 600 }}>A:</span> {body}
+                </div>
+                <button 
+                    onClick={() => this.flipExpanded}
```
Comment (slava000) [inline client/src/components/QuestionsAndAnswers/Answers.jsx]: I think you could just write onClick={this.flipExpanded}, though you may then need to bind "this"

File: client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx [hunk 3]
```diff
@@ -14,6 +14,10 @@ const QABlock = styled.div`
     padding: 20px;
 `;
 
+const BoldSpan = styled.span`
+    font-weight: 600;
+`;
+
 const Question = styled.div`
```
Comment (slava000) [inline client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx:23]: This div has the same properties as BoldSpan, could you consolidate them into one? Or if they will later have additional unique stylings then having two divs makes sense

File: client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx [hunk 4]
```diff
@@ -33,51 +37,49 @@ export default class QuestionAnswer extends React.Component {
             data.results.sort((a, b) => {
                 return b.question_helpfulness - a.question_helpfulness
             });
-            // add expanded to show 2 or more answers
             const qas = data.results.map((qa) => ({ ...qa, expanded: false }))
```
Comment (slava000) [inline client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx]: nice use of rest operator!

File: client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx [hunk 5]
```diff
@@ -33,51 +37,49 @@ export default class QuestionAnswer extends React.Component {
...
             this.setState({ qas })
         });
       }
 
     expandAnswers(qaIndex) {
         const qas = [...this.state.qas];
```
Comment (slava000) [inline client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx]: can you just write const qas = this.state.qas? 
is it necessary to spread an array into an array here?

File: client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx [hunk 6]
```diff
@@ -33,51 +37,49 @@ export default class QuestionAnswer extends React.Component {
...
         this.setState({ qas })
     }
 
     render() {
         return (
             <Container>
                 <h2>Questions and Answers</h2>
-            {this.state.qas.map((qa, qaIndex) => 
-                <QABlock key={qa.id}>
+            {this.state.qas.map((qa, qaIndex) => {
```
Comment (slava000) [inline client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx]: really good mapping and sorting

Comment (slava000) [inline client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx]: this is a really good job / neat map with nested maps and divs. 👏

File: client/src/components/QuestionsAndAnswers/utils/sorting.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            }
+ * @returns sorted array of answers like [{
+                    "id": 5361312,
+                    "body": "What's love got to do with my emotions",
+                    "date": "2022-03-11T00:00:00.000Z",
+                    "answerer_name": "sata5pa3da",
+                    "helpfulness": 2,
+                    "photos": []
+                }]
+ */
```
Comment (slava000) [inline client/src/components/QuestionsAndAnswers/utils/sorting.js:23]: nice, helpful notes

**Review by slava000 — APPROVED** [review]
awesome!

File: client/src/components/QuestionsAndAnswers/utils/sorting.js (hunk 7 above)
Comment (tpminhtam) [reply client/src/components/QuestionsAndAnswers/utils/sorting.js:23]: thanks

File: client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx (hunk 3 above)
Comment (tpminhtam) [reply client/src/components/QuestionsAndAnswers/QuestionAnswer.jsx:23]: good suggestion! i didn't even notice that they were the same, I'm planning to add more styling later so these will be revised later.

**Pull request merged**